podcast/2.15.md

```diff
@@ -0,0 +1,1299 @@
+
+**Sugu Sougoumarane:**
+
+They were all rows in the database. So if I selected all the videos of this user, I got 250,000 rows back. But that wasn't the problem. The problem was that some admin that was curating the index page of YouTube decided to feature that user in YouTube's homepage.
+
+So it's on the front page. Everyone is getting it.
+
+**Beyang Liu:**
+
+It's on the front page.
+
+**Sugu Sougoumarane:**
+
+Which means that every hit on the front page ended up fetching 250,000 videos.
+
+**Beyang Liu:**
+
+Got it. Got it.
+
+**Sugu Sougoumarane:**
+
+So one of the things that we wrote was, well, how do we protect against this? So, the way we did this was again, we had this proxy, right? So this proxy, every time it got a query, it would check if the same query is already executing.
+
+If it is already executing, it won't send it to MySQL, it'll just wait for that result.
+
+**Beyang Liu:**
+
+It's kind of batching the requests.
+
+**Sugu Sougoumarane:**
+
+Or consolidating.
+
+**Beyang Liu:**
+
+Consolidating.
+
+**Sugu Sougoumarane:**
+
+Yeah. So this was an expensive query. At any given point of time, only one of those can run. If the site got spammed with the same query, it won't end up spamming the database. At any given point... if that finishes another one will start, but all other thundering herd requests will wait and will just share the result. So stuff like this, we started adding. And at that time, Vitess had only VT Tablet. Its only job was to protect MySQL, the single MySQL instance, that was-
+
+**Beyang Liu:**
+
+Was is VT Tablet? I'm actually unfamiliar with that.
+
+**Sugu Sougoumarane:**
+
+So VT Tablet was the proxy that we first wrote. That was essentially the Vitess project.
+
+**Beyang Liu:**
+
+I see. Got it.
+
+**Sugu Sougoumarane:**
+
+Now, it is a project that had many other components. So when we wrote it, it was not actually meant to manage sharding. It was only meant to manage your cluster.
+
+Later that evolved into, "Hey, is it ever possible to make the system so generic that someone other than YouTube who wanted to shard could use it." And that was the beginning of what you see as Vitess today.
+
+**Beyang Liu:**
+
+So, how did you convince Google, the company, that this was worth open-sourcing? I mean, my understanding was, it was originally for YouTube, the internal system specifically. Did you have to fight?
+
+**Sugu Sougoumarane:**
+
+Not at all. Actually, it was extremely easy and straightforward. I believe it is still that type of company where if you want to start a project, if you're starting a project and think that this is worthy of being open-sourced, the only thing they want to make sure is that you are not exposing any company secrets that are critical to Google's business. As long as that is not the case, you got approval. Our own decision to open-source was a very selfish decision at that point. If we build this system, I don't want to have to rebuild this if I went somewhere else.
```

Comment:
Check consistency on "open-source" vs "open source" - I think you've got both here and below.